core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -284,11 +292,22 @@ private[spark] class PythonRunner(
         }
         dataOut.flush()
         // Serialized command:
-        dataOut.writeInt(if (rowBased) 1 else 0)
-        dataOut.writeInt(funcs.length)
-        funcs.foreach { f =>
-          dataOut.writeInt(f.command.length)
-          dataOut.write(f.command)
+        if (isUDF) {
+          dataOut.writeInt(1)
+          dataOut.writeInt(funcs.length)
+          funcs.zip(numArgs).foreach { case (fs, numArg) =>
```

Comment:
Since correctness relies on `funcs.length == numArgs.length`, do you mind adding a `require` at the start of the constructor to enforce this?